### ISSUE TITLE
Local/custom sync-engine does not update past initial commit

### ISSUE THREAD

**Issue by ConorIA** [nylas/N1#2596]
##### What operating system are you using?
Manjaro Linux x86-64

##### What version of N1 are you using?
0.4.45-e338050 (Aur package n1-git)

**Bug?**
##### Do you have any third-party plugins installed?
None

##### Is the issue related to a specific email provider (Gmail, Exchange, etc.)?
No

##### Is the issue reproducible with a particular attachment, message, signature, etc?
No 

I am running a small desktop as a server. I have the open source sync-engine installed in a Vagrant on that server, and use nginx to redirect external calls to a given port to localhost:5555.

I can access the sync-engine via curl. 

I use N1 on two computers:
On machine A (desktop/sever), I set up my config.json with env : 'local'
On machine B (laptop/client), I set up my config.json with 'custom'

When I opened N1 on each computer, they performed an initial sync. However, no matter how many times I open/close N1, or how many times I reload vagrant, the N1 clients are not updating. 

I do not think the issue is with sync-engine, beause the desktop, which I set up last night has emails up to yesterday evening and the laptop, which I set up this morning has emails up to 10:29 am. 

On the vagrant machine, everything is running: 
vagrant@vagrant-ubuntu-precise-64:~$ ps aux | grep 'inbox'
```
vagrant   1846  0.0  0.2  25712  1240 ?        Ss   17:06   0:00 SCREEN -A -d -m -U /vagrant/bin/inbox-start
vagrant   1848  0.0  0.2  25712  1236 ?        Ss   17:06   0:00 SCREEN -A -d -m -U /vagrant/bin/inbox-api
vagrant   1849 11.0 17.7 604580 89476 pts/3    Ssl+ 17:06   4:34 inbox-sync-0                   
vagrant   1850  0.0  4.6  85612 23156 pts/4    Ss+  17:06   0:01 inbox-api                    
vagrant   1885 15.8 15.6 264216 78624 pts/4    Sl+  17:06   6:34 inbox-api                    
```

**Comment by bengotow** [comment]
Hi @ConorIA — thanks for reporting this. There are a couple of things you might try to narrow this down.

1 . In N1, open the Developer > Activity Window and click "Requests". See if the app is having trouble making any API requests. 400s and 500s appear in red in that panel.

2. Try making a few API requests to your sync engine manually using `curl` and see if you get new mail. For example, `curl http://ACCOUNT_ID:@localhost:5555/threads`. If you get the new threads in your inbox, it means the problem is in N1. If you don't see new mail, the sync engine may be having trouble.

**Comment by ConorIA** [comment]
@bengotow, thanks for your help. I have now set up the second client and *lo and behold*, it stopped working after the initial sync! There are no new messages appearing in the client. 

Here is the output from activity: 
`20:08:47  RunCopycurl -X GET -H "Accept: message/rfc822" "https://accountIDOne:@my.server.address:XXXX/messages/67z7gndp6nqds5ouq91x0k0bo"
20020:08:47  RunCopycurl -X GET -H "Accept: message/rfc822" "https://accountIDOne:@my.server.address:XXXX/messages/alc8sdihtltmcv62ti1wi3t6q"
20020:08:47  RunCopycurl -X GET -H "Accept: message/rfc822" "https://accountIDOne:@my.server.address:XXXX/messages/7as4aw9qnpy3ng9nn1j7pcdyl"
20020:08:47  RunCopycurl -X GET -H "Accept: message/rfc822" "https://accountIDOne:@my.server.address:XXXX/messages/2y2o3ng41bxtip9i649krkaxd"
20020:08:47  RunCopycurl -X GET -H "Accept: message/rfc822" "https://accountIDOne:@my.server.address:XXXX/messages/zk9vqj7pq4dfk84ycbryrpmt"
20020:08:46  RunCopycurl -X GET -H "Accept: message/rfc822" "https://accountIDOne:@my.server.address:XXXX/messages/9u04krcon6v4t4k3qskcsiy4k"
20020:08:02  RunCopycurl -X GET -H "Accept: message/rfc822" "https://accountIDTwo:@my.server.address:XXXX/messages/a09ejp2f394xg0zx5py9uxoac"
20020:08:02  RunCopycurl -X GET -H "Accept: message/rfc822" "https://accountIDTwo:@my.server.address:XXXX/messages/4ki7kttwoo0jt509hp1x53r6t"
20020:08:02  RunCopycurl -X GET -H "Accept: message/rfc822" "https://accountIDTwo:@my.server.address:XXXX/messages/1cenvdrusch06s7rw061a0n30"
20020:08:02  RunCopycurl -X GET -H "Accept: message/rfc822" "https://accountIDTwo:@my.server.address:XXXX/messages/4gm1uzjbutw0btxheambba4b3"
20020:08:02  RunCopycurl -X GET -H "Accept: message/rfc822" "https://accountIDTwo:@my.server.address:XXXX/messages/866q50a7g4x660ojy4ihow05w"
20020:08:02  RunCopycurl -X GET -H "Accept: message/rfc822" "https://accountIDTwo:@my.server.address:XXXX/messages/5ewkjh74p7voq8zbkbrsetqbg"
20020:08:02  RunCopycurl -X GET -H "Accept: message/rfc822" "https://accountIDTwo:@my.server.address:XXXX/messages/5mff60bs8nmh4mgxl8wd87h9"`

Likewise, when I curl both from the remote machine (over HTTPS, forwarded to 5555 by Nginx) or when I ssh into the host machine and curl on 5555, I can see the latest message (a test message I sent to myself from my phone) does appear in the list of threads, however, it does not appear in the N1 client. 

As far as I can tell, this only started happening once I added the remote client with the "custom" env. Please let me know any other information that could help to debug.

**Comment by ConorIA** [comment]
I seem to have found the issue, but am not sure where it is coming from. In the developer tools console, I am getting a constant stream of:
`/usr/share/nylas/resources/app.asar/internal_packages/worker-sync/lib/nylas-sync-worker.js:38 Backing off after sync failure. Will retry in 3 seconds.`

The error is directly related to this code: 
`BackoffTimer.prototype.backoff = function() {
      this._delay = Math.min(this._delay * 1.7, 5 * 1000 * 60);
      if (!NylasEnv.inSpecMode()) {
        return console.log("Backing off after sync failure. Will retry in " + (Math.floor(this._delay / 1000)) + " seconds.");
      }
    };`

**Comment by ConorIA** [comment]
On the desktop / host, however, there are no requests made at all! And the log doesn't mention any errors. 

![screenshot_20160713_100003](https://cloud.githubusercontent.com/assets/870746/16805973/9cd4c90e-48e0-11e6-9b00-6c96d71ab9f6.png)

**Comment by ConorIA** [comment]
This error appeared on the server machine today. Nothing new on the client. 
![server-screenshot_20160714_114609](https://cloud.githubusercontent.com/assets/870746/16846236/9987ccfe-49b9-11e6-9e8d-295a76f48a0c.png)

**Comment by ConorIA** [comment]
@bengotow, any other debugging hints? As I mentioned, the sync-engine instance has all of the latest messages. I can read them if I use curl as per your suggestion. I don't know why they aren't hitting my client on either of the two machines.

**Comment by ConorIA** [comment]
I rolled back my client to version 4.40 (before the Nylas identity stuff was added). Surprisingly, this version exhibits the same behaviour! So I am guessing the issue is really on the sync-engine side? I have poked though the mysql logs in the vagrant machine, but wouldn't know what to look for, really. 

`root@vagrant-ubuntu-precise-64:/# tail var/log/mysql/*.log
==> var/log/mysql/error.log <==
2016-07-19 17:32:10 1536 [Note] InnoDB: Highest supported file format is Barracuda.
2016-07-19 17:32:10 1536 [Note] InnoDB: 128 rollback segment(s) are active.
2016-07-19 17:32:10 1536 [Note] InnoDB: Waiting for purge to start
2016-07-19 17:32:10 1536 [Note] InnoDB: 5.6.31 started; log sequence number 164561507
2016-07-19 17:32:10 1536 [Note] Server hostname (bind-address): '127.0.0.1'; port: 3306
2016-07-19 17:32:10 1536 [Note]   - '127.0.0.1' resolves to '127.0.0.1';
2016-07-19 17:32:10 1536 [Note] Server socket created on IP: '127.0.0.1'.
2016-07-19 17:32:11 1536 [Note] Event Scheduler: Loaded 0 events
2016-07-19 17:32:11 1536 [Note] /usr/sbin/mysqld: ready for connections.
Version: '5.6.31-log'  socket: '/var/run/mysqld/mysqld.sock'  port: 3306  MySQL Community Server (GPL)

==> var/log/mysql/mysql-slow.log <==
WHERE message.namespace_id = 1 AND message.is_draft = 0 ORDER BY message.received_date DESC 
 LIMIT 4998, 2) AS anon_1 INNER JOIN part ON anon_1.message_id = part.message_id LEFT OUTER JOIN block AS block_1 ON block_1.id = part.block_id ORDER BY anon_1.message_id;
# User@Host: root[root] @ localhost []  Id:    59
# Query_time: 0.116951  Lock_time: 0.000532 Rows_sent: 0  Rows_examined: 12507
SET timestamp=1468953720;
SELECT event.owner2 AS event_owner2, event._description AS event__description, event.type AS event_type, event.public_id AS event_public_id, event.created_at AS event_created_at, event.updated_at AS event_updated_at, event.deleted_at AS event_deleted_at, event.id AS event_id, event.namespace_id AS event_namespace_id, event.calendar_id AS event_calendar_id, event.uid AS event_uid, event.provider_name AS event_provider_name, event.source AS event_source, event.raw_data AS event_raw_data, event.title AS event_title, event.location AS event_location, event.busy AS event_busy, event.`read_only` AS event_read_only, event.reminders AS event_reminders, event.recurrence AS event_recurrence, event.start AS event_start, event.end AS event_end, event.all_day AS event_all_day, event.is_owner AS event_is_owner, event.last_modified AS event_last_modified, event.status AS event_status, event.message_id AS event_message_id, event.participants AS event_participants, event.sequence_number AS event_sequence_number, anon_1.message_id AS anon_1_message_id 
FROM (SELECT message.id AS message_id 
FROM message INNER JOIN thread ON thread.id = message.thread_id 
WHERE message.namespace_id = 1 AND message.is_draft = 0 ORDER BY message.received_date DESC 
 LIMIT 4998, 2) AS anon_1 INNER JOIN event ON anon_1.message_id = event.message_id ORDER BY anon_1.message_id, event.last_modified;`

**Comment by ConorIA** [comment]
I've switched to a docker-based sync-engine install. It seems to have resolved whatever the issue was. I'll re-open if problem re-appears.
